### DISCUSSION TITLE
Chain pubs with delays in cli

### DISCUSSION THREAD

**Discussion by azat385** [emqx/MQTTX#1814] [Q&A]
Hello,
Is there a way to chain the pubs with delays in cli of mqttx?
For example:
pub1
sleep 5
pub2
sleep 2
etc...

Thank you.

**Comment by azat385** [comment]
I mean how to combine pubs and delays in one file using one connection? Use simulate cmd or read-file option?

**Comment by ysfscream** [comment]
Hi! You can chain publications with delays using the `bench pub` command. Here are the key parameters you'll need:

```bash
mqttx bench pub -c 1 -im 5000 -L 3 --file-read messages.txt --split "\n" -t test/topic
```

Key parameters:
- `-c 1`: Use single connection (default is 1000)
- `-im 5000`: Set 5-second delay between messages
- `-L 3`: Limit to publish 3 messages (0 means unlimited)
- `--file-read`: Read messages from file
- `--split "\n"`: Split messages by newline

The messages.txt file format:
```plaintext
{"topic": "sensor/1", "temp": 25.5}
{"topic": "sensor/2", "temp": 26.1}
{"topic": "sensor/3", "temp": 24.8}
```
